ksqldb-tools/src/main/java/io/confluent/ksql/tools/migrations/commands/ApplyMigrationCommand.java

```diff
@@ -56,7 +77,201 @@
+    final List<Migration> migrations = getAllMigrations(migrationsDir).stream()
+        .filter(migration -> {
+          if (version > 0) {
+            return migration.getVersion() <= version && migration.getVersion() >= minimumVersion;
+          } else {
+            return migration.getVersion() >= minimumVersion;
+          }
+        })
+        .collect(Collectors.toList());
+
+    if (migrations.size() == 0) {
+      LOGGER.error("No eligible migrations found.");
+      return false;
+    } else {
+      LOGGER.info(migrations.size() + " migration files loaded.");
+    }
+
+    for (Migration migration : migrations) {
+      if (!applyMigration(config, ksqlClient, migration, clock, previous)) {
+        return false;
+      }
+      previous = Integer.toString(migration.getVersion());
+    }
+    return true;
+  }
+
+  private boolean applyMigration(
+      final MigrationConfig config,
+      final Client ksqlClient,
+      final Migration migration,
+      final Clock clock,
+      final String previous
+  ) {
+    LOGGER.info("Applying " + migration.getName() + " version " + migration.getVersion());
+    LOGGER.info(migration.getCommand());
+
+    if (dryRun) {
+      return true;
+    }
+
+    if (!verifyMigrated(config, ksqlClient, previous, MAX_RETRIES)) {
+      LOGGER.error("Failed to verify status of version " + previous);
+      return false;
+    }
+
+    final String executionStart = Long.toString(clock.millis());
+
+    if (
+        !updateState(config, ksqlClient, MigrationState.RUNNING,
+            executionStart, migration, clock, previous)
+    ) {
+      return false;
+    }
+
+    try {
+      final List<String> commands = Arrays.stream(migration.getCommand().split(";"))
+          .filter(s -> s.length() > 1)
+          .collect(Collectors.toList());
+      for (final String command : commands) {
+        ksqlClient.executeStatement(command + ";").get();
+      }
+    } catch (InterruptedException | ExecutionException e) {
+      LOGGER.error(e.getMessage());
+      updateState(config, ksqlClient, MigrationState.ERROR,
+          executionStart, migration, clock, previous);
+      return false;
+    }
+    updateState(config, ksqlClient, MigrationState.MIGRATED,
+        executionStart, migration, clock, previous);
+    LOGGER.info("Successfully migrated");
+    return true;
+  }
+
+  private boolean verifyMigrated(
+      final MigrationConfig config,
+      final Client ksqlClient,
+      final String version,
+      final int retries
+  ) {
+    if (version.equals(MetadataUtil.NONE_VERSION)) {
+      return true;
+    }
+    try {
+      RetryUtil.retryWithBackoff(
+          retries,
+          1000,
+          1000,
+          () -> {
+            if (!MetadataUtil.getInfoForVersion(version, config, ksqlClient)
+                .getState().equals(MigrationState.MIGRATED)) {
+              throw new MigrationException("Could not verify status of version " + version);
```

Comment:
nit: make the error message more specific, something like `Expected status MIGRATED for version <version>. Got <other status>`?